### PR TITLE
Begin migrating tools to NNBD

### PR DESCRIPTION
**Pull request by stuartmorgan**
- Updates dependencies to null-safe versions
- Migrates common.dart (which doesn't depend on anything)
- Migrates common_tests.dart and its one dependency, utils.dart
- Adds build_runner for Mockito mock generation
- Adds a new utility methods for getting arguments that handle both the casting and the removal of nullability to address a common problematic pattern while migrating code.
  - Converts all files, not just the migrated ones, to those new helpers.

Migrating common.dart and utils.dart should unblock a command-by-command migration to null safety.

Reverts the separate of podspect lints into a step that doesn't do a Flutter upgrade
(https://github.com/flutter/plugins/pull/3700) because without that step we had a
version of Dart too old to run null-safe tooling.

First step of https://github.com/flutter/flutter/issues/81912

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`. See [plugin_tool format](../script/tool/README.md#format-code))
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: script/tool/lib/src/common.dart [hunk 1]
```diff
@@ -584,7 +596,7 @@ class PubVersionFinder {
 
   /// Get the package version on pub.
   Future<PubVersionFinderResponse> getPackageVersion(
-      {@required String package}) async {
+      {required String? package}) async {
```
Comment (cyanglaz) [inline script/tool/lib/src/common.dart:599]: can be `required String package` and remove the below assert for != null

**Review by cyanglaz — CHANGES_REQUESTED** [review]

File: script/tool/lib/src/common.dart [hunk 2]
```diff
@@ -87,7 +86,7 @@ bool pluginSupportsPlatform(
       platform == kMacos ||
       platform == kWindows ||
       platform == kLinux);
-  if (entity == null || entity is! Directory) {
+  if (entity is! Directory) {
```
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply script/tool/lib/src/common.dart:89]: This is the very beginning of the method :)

File: script/tool/lib/src/common.dart (hunk 1 above)
Comment (stuartmorgan) [reply script/tool/lib/src/common.dart:599]: Done.

**Review by cyanglaz — APPROVED** [review]
LGTM

**Pull request merged**